src/parseSetupScript.ts

```diff
@@ -0,0 +1,41 @@
+// import ts from 'typescript';
+
+export default async function parseSetupScript(script: HTMLScriptElement, scriptCode: string) {
+    const isTypeScript = (script?.attributes as any)?.lang?.value === 'ts'
```

Comment:
I suggest that the processing of `setup` and `typescript` can be separated here

Reply:
Yeah that can be done